consensus/blake3pow/consensus.go

```diff
@@ -340,6 +339,49 @@ func CalcDifficulty(config *params.ChainConfig, time uint64, parent *types.Heade
 	return x
 }
 
+// CalcDifficultyAtIndex is the difficulty adjustment algorithm. It returns
+// the difficulty that a new block should have when created at time
+// given the parent block's time and difficulty.
+// NOTE: This is essentially the Ethereum DAA, without a 'difficulty bomb'
+func CalcDifficultyAtIndex(config *params.ChainConfig, time uint64, parent *types.Header, context int) *big.Int {
```

Comment:
Please do not pass in `context`

Reply:
If there is a scenario where we need to calculate difficulty adjustments for other contexts, then follow the variadic arg paradigm used everywhere else. See the header accessors for example: https://github.com/dominant-strategies/go-quai/blob/c1d1e2dc456fcef71d098ed3befecd9d143f8f26/core/types/block.go#L201